### PR TITLE
Install openstack

### PR DESCRIPTION
**Pull request by cgilmour**
New feature for the installer to deploy openstack using packages, and configure/enable Romana.
Supports both Ubuntu 14.04 and Centos 7.

Tested on Vagrant by completing the install and running:
```bash
nova boot --flavor m1.micro --image cirros-0.3.4-x86_64-uec --key-name shared-key --nic net-id=$(neutron net-show romana -Fid -f value) --meta romanaSegment=backend b1 --availability-zone=nova:cgilmour-controller
nova boot --flavor m1.micro --image cirros-0.3.4-x86_64-uec --key-name shared-key --nic net-id=$(neutron net-show romana -Fid -f value) --meta romanaSegment=backend b1 --availability-zone=nova:cgilmour-compute01
```
This results in two instances running
```sh-session
[centos@cgilmour-controller ~]$ nova list
+--------------------------------------+------+--------+------------+-------------+-----------------+
| ID                                   | Name | Status | Task State | Power State | Networks        |
+--------------------------------------+------+--------+------------+-------------+-----------------+
| beacbdcf-d71c-4a77-8434-7529563c9d09 | b1   | ACTIVE | -          | Running     | romana=10.0.2.3 |
| d6ab5f31-db84-4101-b220-c23738316119 | b1   | ACTIVE | -          | Running     | romana=10.1.2.3 |
+--------------------------------------+------+--------+------------+-------------+-----------------+
```
SSH and Ping between the instances works.
```sh-session
[centos@cgilmour-controller ~]$ ssh cirros@10.0.2.3
$ ping 10.1.2.3
PING 10.1.2.3 (10.1.2.3): 56 data bytes
64 bytes from 10.1.2.3: seq=0 ttl=62 time=3.513 ms
64 bytes from 10.1.2.3: seq=1 ttl=62 time=1.270 ms
^C
--- 10.1.2.3 ping statistics ---
2 packets transmitted, 2 packets received, 0% packet loss
round-trip min/avg/max = 1.270/2.391/3.513 ms
$ ssh 10.1.2.3
cirros@10.1.2.3's password: 
$ 
```

### CODE REVIEW

File: romana-install/roles/stack/openstack/install-controller/tasks/neutron.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- include: neutron_db.yml
+
+- name: Install neutron post-install script
+  template: src="neutron-post-install.sh" dest="/var/tmp/neutron-post-install.sh" mode=0755
+
+- name: Execute neutron post-install script
+  shell: /var/tmp/neutron-post-install.sh
+
+- name: Install networking-romana module
+  pip: name="git+https://github.com/romana/networking-romana@realign-requirements" editable=false
```
Comment (cgilmour) [inline romana-install/roles/stack/openstack/install-controller/tasks/neutron.yml]: This needs fixing after merging romana/networking-romana#13
(It should end with `@stable/liberty`)

File: romana-install/config.devstack.yml [hunk 2]
```diff
@@ -17,3 +17,7 @@
 - hosts: devstack_computes
   roles:
     - stack/devstack/install-compute
+
+- hosts: devstack_controller
+  roles:
+    - stack/devstack/postinstall
```
Comment (jbrendel) [inline romana-install/config.devstack.yml]: 'postinstall' only applies to the controller?

File: romana-install/group_vars/vagrant [hunk 3]
```diff
@@ -1,9 +1,11 @@
 ram_size:
   controller:
     devstack: 4096
+    openstack: 4096
     kubernetes: 2048
   compute:
     devstack: 1024
+    openstack: 1024
```
Comment (jbrendel) [inline romana-install/group_vars/vagrant]: Weird that kubernetes nodes get more memory than openstack and devstack nodes, right?

File: romana-install/roles/romana/install-agent/tasks/network.yml [hunk 4]
```diff
@@ -21,10 +21,12 @@
...
     - POSTROUTING -t mangle -p udp --dport 68 -j CHECKSUM --checksum-fill
 
+# Ensure IP forwarding is enabled.
 # Ensure ARP requests/replies can be handled across interfaces.
 # This is so that endpoints behind TAP interfaces can reach the host/gateway.
-- name: Enable kernel proxy_arp setting
+- name: Enable kernel ip forwarding and proxy_arp settings
   shell: if [ $(sysctl -n "{{ item }}") != "1" ]; then sysctl "{{ item }}"=1; fi
   with_items:
+    - net.ipv4.ip_forward
```
Comment (jbrendel) [inline romana-install/roles/romana/install-agent/tasks/network.yml]: How did this ever work before? Was forwarding not enabled via our installer?

Comment (cgilmour) [inline romana-install/roles/romana/install-agent/tasks/network.yml]: Devstack does it as part of installing nova. Not sure for kubernetes.

File: romana-install/group_vars/vagrant (hunk 3 above)
Comment (cgilmour) [inline romana-install/group_vars/vagrant]: Yup. It related to doing a kubernetes demo and running out of RAM.

File: romana-install/config.devstack.yml (hunk 2 above)
Comment (cgilmour) [inline romana-install/config.devstack.yml]: It's things that need to be done once to prepare the system (creating network, adding keypair etc). It doesn't need doing on compute nodes. Any node would do, but controller seemed more sensible.

**Pull request closed without merging**